src/components/pages/Publish/index.tsx

```diff
@@ -137,73 +164,78 @@ export default function PublishPage({
   return isInPurgatory && purgatoryData ? null : (
     <Formik
       initialValues={
-        publishType === TypeOfPublish.dataset
-          ? initialValues
-          : initialValuesAlgorithm
+        publishType === 'dataset' ? initialValues : initialValuesAlgorithm
       }
       initialStatus="empty"
       validationSchema={
-        publishType === TypeOfPublish.dataset
-          ? validationSchema
-          : validationSchemaAlgorithm
+        publishType === 'dataset' ? validationSchema : validationSchemaAlgorithm
       }
       onSubmit={async (values, { resetForm }) => {
         // move user's focus to top of screen
         window.scrollTo({ top: 0, left: 0, behavior: 'smooth' })
         // kick off publishing
-        publishType === TypeOfPublish.dataset
+        publishType === 'dataset'
           ? await handleSubmit(values, resetForm)
           : await handleAlgorithmSubmit(values, resetForm)
       }}
     >
-      {({ values }) => (
-        <>
-          <Persist name={formName} ignoreFields={['isSubmitting']} />
-
-          {hasFeedback ? (
-            <MetadataFeedback
-              title={title}
-              error={error}
-              success={success}
-              loading={publishStepText}
-              setError={setError}
-              successAction={{
-                name: 'Go to data set →',
-                to: `/asset/${did}`
-              }}
+      {({ values }) => {
+        const tabs = [
+          {
+            title: 'Data Set',
+            content: <TabContent values={values} publishType={publishType} />
+          },
+          {
+            title: 'Algorithm',
+            content: <TabContent values={values} publishType={publishType} />
+          }
+        ]
+
+        return (
+          <>
+            <Persist
+              name={
+                publishType === 'dataset'
+                  ? formNameDatasets
+                  : formNameAlgorithms
```

Comment:
Unfortunately this is not enough, I tried it also today, you should also remove the resetForm call from the FormAlgoPublish.ts (https://github.com/oceanprotocol/market/blob/e157278e37a82305852581a89f28b00c96cf953f/src/components/pages/Publish/FormAlgoPublish.tsx#L60)

It's something strange i guess it's related to formik values, somehow the values from localStorage is override when switching tabs and the other form mounts. Also strange why there are stored in the initial values for the algorithm form the same values as for the dataset even though we pass other values there.
<img width="1657" alt="Screenshot 2021-02-22 at 22 30 17" src="https://user-images.githubusercontent.com/13447142/108766524-ef2c5380-755d-11eb-840c-eb9902291d54.png">

Reply:
good findings, that's worth looking at individually in publish PR or as another PR to fully restore this behavior. But we should not get blocked by this, would be also fine moving this into a new issue to deal with it later